crates/lox_core/src/ephemeris/daf_spk/parser.rs

```diff
@@ -398,10 +398,12 @@ pub fn parse_daf_spk(full_input: &[u8]) -> Result<Spk, DafSpkError> {
         .into_iter()
         .flatten()
         .fold(HashMap::new(), |mut map, segment| {
+            // An SPK file may contain any number of segments. A single file may contain overlapping segments:
+            // segments containing data for the same body over a common interval. When this happens, the
+            // latest segment in a file supersedes any competing segments earlier in the file.
             map.entry(segment.center_id)
                 .or_default()
-                .entry(segment.target_id)
-                .or_insert(segment);
+                .insert(segment.target_id, segment);
```

Comment:
@AngusGMorrison @helgee  can someone cross check my thinking here?

Reply:
Your comment is based on [this part](https://naif.jpl.nasa.gov/pub/naif/toolkit_docs/C/req/spk.html#Segment%20Order%20and%20Priority) of the docs?

The SPK kernels that we will be using almost exclusively (DE440, DE430, DE405) all have a single segment per center/target body combination but the long-term kernels would have more (and as an added complication are split across multiple files). Just realized that I did not consider this case in JPLEphemeris.jl as well.

Thus, a proper design could be to model the data structure like this `HashMap<BodyId, HashMap<BodyId, Vec<SpkSegment>>>`, i.e. save the segments indiscriminately and put the segment selection logic further downstream.

But I would also be fine with leaving a `TODO` and moving on.

Reply:
Changed as `Vec<SpkSegment>`.